ext/OptimizationEnzymeExt.jl

```diff
@@ -141,25 +148,27 @@ function Optimization.instantiate_function(f::OptimizationFunction{true},
     cache::Optimization.ReInitCache,
     adtype::AutoEnzyme,
     num_cons = 0)
-    _f = (f, θ, args...) -> first(f(θ, cache.p, args...))
+    p = cache.p
 
     if f.grad === nothing
         function grad(res, θ, args...)
             res .= zero(eltype(res))
             Enzyme.autodiff(Enzyme.Reverse,
-                Const(_f),
+                Const(firstapply),
                 Const(f.f),
                 Enzyme.Duplicated(θ, res),
+                Const(p),
                 args...)
         end
     else
         grad = (G, θ, args...) -> f.grad(G, θ, p, args...)
     end
 
     if f.hess === nothing
-        function g(θ, bθ, _f, f, args...)
-            Enzyme.autodiff_deferred(Enzyme.Reverse, Const(_f), Const(f),
+        function g(θ, bθ, f, p, args...)
+            Enzyme.autodiff_deferred(Enzyme.Reverse, Const(firstapply), Const(f),
```

Comment:
```suggestion
            Enzyme.autodiff_deferred(Enzyme.Reverse, Const(firstapply), Active, Const(f),
```

Reply:
what is the active here for?

Reply:
Enzyme will autodeduce the return type from the type, if its inferrable. However, really type unstable coded may have its return type deduced as any, which is duplicated, and not supported in reverse mode by default yet.